roadmap/implementers-guide/src/runtime/scheduler.md

```diff
@@ -1,3 +1,227 @@
+- Top level responsibilities of the scheduler
+- Interfaces
+- Considerations async backing (see ticket)
+
+# Interfaces for parathreads
+
+## User Interface
+
+The entry point for the user is twofold:
+
+1. Registering a PVF with some deposit
+2. Actually bidding for a slot.
+
+### Registering a PVF with some deposit
+
+This can very likely be equal or very similar to what we have for parachains.
+
+TODO: Figure out how this works for parachains or whether there even is a
+separate deposit for the PVF. I believe parachains just have locked DOTs for
+their lease period, which also covers PVF storage.
+
+### Bidding for a slot/Auctioning System
+
+#### Goals
+
+1. Price finding
+2. Be efficient ... we are talking about slots for single blocks, the overhead
+   of assignment should be minimal.
+3. Maximize utilization: Avoid empty slots, as long as there is demand from
+   parathread willing to pay the minimum price.
+4. DoS resistance: We should not be able to get flooded with dubious bids, that
+   just fill up runtime resources.
+5. Lifetime/Update: Bids should have some lifetime or alternatively need to be
+   upgradeable. E.g. someone puts in a bid, but it is too low and is constantly
+   outbid by other bids ... eventually we either no longer care ... would like
+   to remove the bid or increase it, so it will finally make it.
```

Comment:
I think cancel and then new bid would be the cleanest way. We could give the user an interface to up their bid but internally, it could still be cancel(and remove) and add a new big with the new price.

Reply:
Yes, arrived at the same. Likely no explicit cancel, but just rather low lifetime of a bid, e.g. 10 blocks.

Reply:
I suggest we don't expose the inner workings of the system to parathreads. Tipically, parathreads would want to buy some amount of block space, say X blocks, starting at relay block N, but no later than block M. [N,M] clearly represents a lifetime, and they would know if they got the slots by doing what they already do, looking up core assignments in the relay chain state. If we are at block M+1 and we did not get ourselves assigned to a core (X times), then we know that our price was too low, and our order was not filled, or just partially filled.

All of the bids would live in the order book, and should be exposed via some well known keys in the relay chain state, or we could have a runtime API to fetch them.

Reply:
If parathreads want to change their bid, they can submit the same extrinsic with a different price and pay an additional small fixed fee for doing so.